Microsoft/WebTemplateStudio
# Dandua98/azure func zip

This PR introduces support for deploying azure functions through the extension using credentials from _azure-auth_. While the functions introduced by the module would be used by wizard through the messages API directly, they're registered as commands for now to support testing. 

This PR closes Azure DevOps tickets **23195, 23203** and **23211**.


_**Testing Notes:**_

-  Set the subscription Id for Giv.Hackathon (_recommended_) as an environment variable (SUBSCRIPTION_ID). You might need to restart vscode and rebuild to have the new environment variable override vscode cache. Note that this is temporary for testing as mentioned earlier. The wizard would actually pass a JSON of the selections to the function
- Make sure you're signed in to _azure-accounts_ extension! The accounts module doesn't handle redirecting you to sign in if you aren't signed in
- In the hardcoded selections JSON in _extension.ts_, choose a unique functionApp name! Note that the wizard would directly call _checkFunctionAppName_ to check for validity and then call _createFunctionApp_. Since the wizard azure functions screen isn't ready yet, this is a temporary way to test.
- You can change the _resourceGroupName_ if required to another valid resource group under the subscription
- Note that invalid behavior would throw errors with the expectancy that the wizard would handle them.
